### DISCUSSION TITLE
Can I read and parse static json file with cloudflare-workers?

### DISCUSSION THREAD

**Discussion by hectorAguero** [honojs/hono#1375] [Q&A]
I had a json list file in my static folder that I use to search an item by id(basically like a table/document but small).

With **Deno.readTextFile** or **node:fs/promises**, is easy to do, but cloudflare workers doesn't support that.

Based on the docs I figure out I can do with serve static file, but I need to search inside the json also, so I need to retrieve the file, parse, search and return the item.

Edit: add code example

```ts
pokemons.get('/:id', async (context) => {
    const id = parseInt(context.req.param('id'));
    const { language } = context.req.query();
    //  const file = await Deno.readTextFile(`pokemons_${ language || 'en'}.json`);
    const file =  app.get(`*`, serveStatic({ path: `pokemons_${ language || 'en'}.json` }));
    // Parse the JSON into JavaScript object
    const pokemons = JSON.parse(file);
    // Find the Pokemon with the matching id
    const pokemon = pokemons.find((pokemon: Pokemon) => pokemon.id === id);
    if (!pokemon) {
      return context.json({ error: 'Pokemon not found' }, 404);
    } else {
      return context.json(pokemon);
    }
  });
```

**Comment by yusukebe** [comment]
Hi @hectorAguero!

You've raised an excellent point, and I've been contemplating the best approach to address it. Indeed, unlike Deno, Bunt, and Node.js, Cloudflare Workers doesn't have a conventional "file-system". This certainly complicates matters.

I've experimented with the "Workers Sites" before, which serves assets for Cloudflare Workers via Cloudflare KV. Here's a relevant example for reference:

https://github.com/yusukebe/ramen-api

In this project, content JSON files are stored in the Workers Sites (KV), and content is retrieved using `getContentFromKVAsset()` from Hono's utilities. While this feels like an good "workaround", there's a catch. Cloudflare has hinted at deprecating "Workers Sites".

https://developers.cloudflare.com/workers/configuration/sites/

They now recommend Cloudflare Pages over Workers Sites. This leaves us with using KV, R2, or bundling everything into a single JavaScript file. None of these methods seem ideal. I've also contemplated creating a blog system that reads Markdown files, but I anticipate encountering a similar challenge.

Let's keep this thread open and active. I'll consult with other developers at Cloudflare.

**Comment by nicholasglazer** [comment]
this is written in Hono docs 
https://hono.dev/docs/getting-started/cloudflare-workers#serve-static-files
